persistence/src/test/java/cz/muni/fi/pa165/mushrooms/dao/VisitDaoTest.java

```diff
@@ -0,0 +1,384 @@
+        testVisit.setDate(LocalDate.of(2017,2,5));
+        testVisit.setNote("note for the forest");
+
+        assertThatThrownBy(() -> visitDao.update(testVisit)).isInstanceOf(ConstraintViolationException.class);
+    }
+
+    @Test
+    public void update_visitWithNullForest() {
+        MushroomHunter testHunter = new MushroomHunter();
+        testHunter.setFirstName("Maradona");
+        testHunter.setSurname("Sandokan");
+        testHunter.setUserNickname("bidG");
+
+        em.persist(testHunter);
+
+        Visit testVisit = new Visit();
+        testVisit.setHunter(testHunter);
+        testVisit.setForest(null);
+        testVisit.setDate(LocalDate.of(2017,2,5));
+        testVisit.setNote("note for the forest");
+
+        assertThatThrownBy(() -> visitDao.update(testVisit)).isInstanceOf(ConstraintViolationException.class);
+
+    }
+
+    @Test
+    public void delete_validVisit() throws Exception {
+        int sizeBeforeCreate = visitDao.findAll().size();
+
+        visitDao.delete(visit1);
+        em.flush();
+
+        assertThat(visitDao.findAll().size()).isEqualTo(sizeBeforeCreate - 1);
+        assertThat(visitDao.findAll()).containsExactlyInAnyOrder(visit2);
+    }
+
+    @Test
+    public void delete_nullVisit() throws Exception {
+        assertThatThrownBy(() -> visitDao.delete(null)).hasRootCauseInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void delete_nonExistingVisit() {
+        int sizeBeforeCreate = visitDao.findAll().size();
+        Forest testForest = new Forest();
+        testForest.setName("Test Forest");
+
+        MushroomHunter testHunter = new MushroomHunter();
+        testHunter.setFirstName("Maradona");
+        testHunter.setSurname("Sandokan");
+        testHunter.setUserNickname("bidG");
+
+        em.persist(testForest);
+        em.persist(testHunter);
+
+        Visit testVisit = new Visit();
+        testVisit.setHunter(testHunter);
+        testVisit.setForest(testForest);
+        testVisit.setDate(LocalDate.of(2017,2,5));
+
+        visitDao.delete(testVisit);
+        em.flush();
+
+        assertThat(visitDao.findAll().size()).isEqualTo(sizeBeforeCreate);
+        assertThat(visitDao.findAll()).containsExactlyInAnyOrder(visit1,visit2);
+    }
+
+    @Test
+    public void findAll_visits() throws Exception {
+        assertThat(visitDao.findAll()).containsExactlyInAnyOrder(visit1,visit2);
+    }
+
+    @Test
+    public void findById_nullId() throws Exception {
+        assertThatThrownBy(() -> visitDao.findById(null)).hasRootCauseExactlyInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void findById_validId() throws Exception {
+        Visit testForest = visitDao.findById(visit1.getId());
+        assertThat(testForest).isNotNull();
+        assertThat(testForest).isEqualTo(visit1);
+    }
+
+    @Test
+    public void findById_invalidId() throws Exception {
+        Long invalidID = new Long(64000);
```

Comment:
"random ID" - works, but is fragile and dependent on the database implementation. visit1.getId()+visit2.getId() should be safer.

Reply:
ok, changed in the code